examples/test_curve.py

```diff
@@ -154,85 +156,32 @@ def make_bg_model_two_groups():
-
-    exclusion_mask = ExclusionMask.read('$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits')
-    exclusion_mask = exclusion_mask.reproject(refheader=refheader)
+    offset_band = Angle([0, 2.49], 'deg')
+    data_store = DataStore.from_dir('/Users/jouvin/Desktop/these/temp/bg_model_image/')
 
     # TODO: fix `binarize` implementation
     # exclusion_mask = exclusion_mask.binarize()
+    image = SkyMap.empty(nxpix=1000, nypix=1000, binsz=0.01, xref=center.l.deg,
+                         yref=center.b.deg, proj='TAN', coordsys='GAL')
 
-    for obs_id in data_store.obs_table['OBS_ID'][:2]:
-
-        obs = data_store.obs(obs_id=obs_id)
-        counts_image = SkyMap.empty_like(counts_image_total)
-        bkg_image = SkyMap.empty_like(counts_image_total)
-
-        log.info('Processing OBS_ID = {}'.format(obs_id))
-
-        log.info('Processing EVENTS')
-        events = obs.events
-        log.debug('Number of events before selection: {}'.format(len(events)))
-        events = events.select_energy(energy_band)
-        log.debug('Number of events after selection: {}'.format(len(events)))
-        counts_image.fill(events=events)
-
-        log.info('Processing BKG')
-        center = obs.pointing_radec.galactic
-        livetime = obs.observation_live_time_duration
-        solid_angle = Angle(0.01, "deg") ** 2
-
-        # TODO: this is broken at the moment ... the output image is full of NaNs
-        table = obs.bkg.acceptance_curve_in_energy_band(energy_band=energy_band)
-        bkg_hdu = fill_acceptance_image(refheader, center, table["offset"], table["Acceptance"])
-        bkg = Quantity(bkg_hdu.data, table["Acceptance"].unit) * solid_angle * livetime
-        bkg = bkg.decompose()
-
-        counts_sum = np.sum(counts_image.data * exclusion_mask.data)
-        bkg_sum = np.sum(bkg_image.data * exclusion_mask.data)
-        scale = counts_sum / bkg_sum
-        bkg_image.data = scale * bkg
-
-        log.debug('scale = {}'.format(scale))
-        a = np.sum(bkg_image.data * exclusion_mask.data)
-        log.debug('a = {}'.format(a))
-        log.debug('counts_sum = {}'.format(counts_sum))
-
-        # Stack counts and background in total skymaps
-        counts_image_total.data += counts_image.data
-        bkg_image_total.data += bkg_image.data
-
-        import IPython; IPython.embed(); 1/0
+    refheader = image.to_image_hdu().header
+    exclusion_mask = ExclusionMask.read('$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits')
+    exclusion_mask = exclusion_mask.reproject(refheader=refheader)
+    images = ImageAnalysis(image, energy_band=energy_band, offset_band=offset_band, data_store=data_store,
+                           exclusion_mask=exclusion_mask)
 
-    maps = SkyMapCollection()
-    maps['counts'] = counts_image_total
-    maps['bkg'] = bkg_image_total
-    maps['exclusion'] = exclusion_mask
+    exclusion_mask = ExclusionMask.read('$GAMMAPY_EXTRA/datasets/exclusion_masks/tevcat_exclusion.fits')
```

Comment:
Why read the same exclusion mask a second time here?
Should this and the following line be removed?